node/node.go

```diff
@@ -10,38 +10,43 @@ import (
 	"github.com/nats-io/nats.go"
 )
 
-type NexNode struct {
+type Node interface {
+	Validate() error
+	Start()
+}
+
+type nexNode struct {
 	nc *nats.Conn
 
 	logger *slog.Logger
 }
 
-type NexOption func(*NexNode)
+type NexOption func(*nexNode)
```

Comment:
I think this should be `func(*NodeOptions)`, and the node options are a separate struct from the nex node.